My SQL Issue

It was build without any issue but when tring to up the mysql it'll exit with code 1. bellow is the log.

 ✔ Container laradock-mysql-1  Created                                                                                                           0.0s
Attaching to mysql-1
mysql-1  | 2024-07-04 08:45:58+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 9.0.0-1.el9 started.
mysql-1  | 2024-07-04 08:45:58+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
mysql-1  | 2024-07-04 08:45:58+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 9.0.0-1.el9 started.
mysql-1  | '/var/lib/mysql/mysql.sock' -> '/var/run/mysqld/mysqld.sock'
mysql-1  | 2024-07-04T08:45:58.882484Z 0 [System] [MY-015015] [Server] MySQL Server - start.
mysql-1  | 2024-07-04T08:45:59.151218Z 0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
mysql-1  | 2024-07-04T08:45:59.153069Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 9.0.0) starting as process 1
mysql-1  | 2024-07-04T08:45:59.154729Z 0 [Warning] [MY-013242] [Server] --character-set-server: 'utf8' is currently an alias for the character set UTF8MB3, but will be an alias for UTF8MB4 in a future release. Please consider using UTF8MB4 in order to be unambiguous.
mysql-1  | 2024-07-04T08:45:59.161123Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
mysql-1  | 2024-07-04T08:45:59.611784Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
mysql-1  | mysqld: Table 'mysql.plugin' doesn't exist
mysql-1  | 2024-07-04T08:45:59.789324Z 0 [ERROR] [MY-010735] [Server] Could not open the mysql.plugin table. Please perform the MySQL upgrade procedure.
mysql-1  | 2024-07-04T08:45:59.789709Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
mysql-1  | 2024-07-04T08:45:59.789978Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
mysql-1  | 2024-07-04T08:45:59.790239Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
mysql-1  | 2024-07-04T08:45:59.790604Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
mysql-1  | 2024-07-04T08:45:59.790852Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
mysql-1  | 2024-07-04T08:45:59.791088Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
mysql-1  | 2024-07-04T08:45:59.791317Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
mysql-1  | 2024-07-04T08:45:59.850014Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
mysql-1  | 2024-07-04T08:45:59.925321Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
mysql-1  | 2024-07-04T08:45:59.950145Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
mysql-1  | 2024-07-04T08:45:59.950235Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
mysql-1  | 2024-07-04T08:45:59.960992Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
mysql-1  | 2024-07-04T08:45:59.961729Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
mysql-1  | 2024-07-04T08:45:59.962087Z 0 [ERROR] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-001146 - Table 'mysql.component' doesn't exist
mysql-1  | 2024-07-04T08:45:59.962117Z 0 [Warning] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-003543 - The mysql.component table is missing or has an incorrect definition.
mysql-1  | 2024-07-04T08:45:59.962371Z 0 [ERROR] [MY-000067] [Server] unknown variable 'mysql_native_password=on else echo default-authentication-plugin=mysql_native_password'.
mysql-1  | 2024-07-04T08:45:59.964077Z 0 [ERROR] [MY-010119] [Server] Aborting
mysql-1  | 2024-07-04T08:46:01.034251Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 9.0.0)  MySQL Community Server - GPL.
mysql-1  | 2024-07-04T08:46:01.034270Z 0 [System] [MY-015016] [Server] MySQL Server - end.
mysql-1 exited with code 1

更换为8.0+版本的mysql也会出现bug，下方是log

2024-07-05 14:12:09 2024-07-05 14:12:09+08:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.38-1.el9 started.
2024-07-05 14:12:10 2024-07-05 14:12:10+08:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-07-05 14:12:10 2024-07-05 14:12:10+08:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.38-1.el9 started.
2024-07-05 14:12:10 '/var/lib/mysql/mysql.sock' -> '/var/run/mysqld/mysqld.sock'
2024-07-05 14:12:10 2024-07-05T06:12:10.599715Z 0 [Warning] [MY-011068] [Server] The syntax '--skip-host-cache' is deprecated and will be removed in a future release. Please use SET GLOBAL host_cache_size=0 instead.
2024-07-05 14:12:10 2024-07-05T06:12:10.599763Z 0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2024-07-05 14:12:10 2024-07-05T06:12:10.603267Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.38) starting as process 1
2024-07-05 14:12:10 2024-07-05T06:12:10.604129Z 0 [Warning] [MY-013242] [Server] --character-set-server: 'utf8' is currently an alias for the character set UTF8MB3, but will be an alias for UTF8MB4 in a future release. Please consider using UTF8MB4 in order to be unambiguous.
2024-07-05 14:12:10 2024-07-05T06:12:10.606659Z 0 [Warning] [MY-010159] [Server] Setting lower_case_table_names=2 because file system for /var/lib/mysql/ is case insensitive
2024-07-05 14:12:10 2024-07-05T06:12:10.613048Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-07-05 14:12:10 2024-07-05T06:12:10.880249Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2024-07-05 14:12:10 mysqld: Table 'mysql.plugin' doesn't exist
2024-07-05 14:12:10 2024-07-05T06:12:10.976544Z 0 [ERROR] [MY-010735] [Server] Could not open the mysql.plugin table. Please perform the MySQL upgrade procedure.
2024-07-05 14:12:10 2024-07-05T06:12:10.976811Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-05 14:12:10 2024-07-05T06:12:10.976976Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-05 14:12:10 2024-07-05T06:12:10.977129Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-05 14:12:10 2024-07-05T06:12:10.977384Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-05 14:12:10 2024-07-05T06:12:10.977584Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-05 14:12:10 2024-07-05T06:12:10.978243Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-05 14:12:10 2024-07-05T06:12:10.978608Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-05 14:12:11 2024-07-05T06:12:11.030700Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-07-05 14:12:11 2024-07-05T06:12:11.104352Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-07-05 14:12:11 2024-07-05T06:12:11.114037Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2024-07-05 14:12:11 2024-07-05T06:12:11.114410Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2024-07-05 14:12:11 2024-07-05T06:12:11.122914Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
2024-07-05 14:12:11 2024-07-05T06:12:11.123323Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-05 14:12:11 2024-07-05T06:12:11.123558Z 0 [ERROR] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-001146 - Table 'mysql.component' doesn't exist
2024-07-05 14:12:11 2024-07-05T06:12:11.123611Z 0 [Warning] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-003543 - The mysql.component table is missing or has an incorrect definition.
2024-07-05 14:12:11 2024-07-05T06:12:11.123707Z 0 [ERROR] [MY-000067] [Server] unknown variable 'mysql_native_password=on else echo default-authentication-plugin=mysql_native_password'.
2024-07-05 14:12:11 2024-07-05T06:12:11.124093Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-07-05 14:12:12 2024-07-05T06:12:12.248991Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.38)  MySQL Community Server - GPL.

I also have the same problem I tried every solution provided in this open issue and other different github issues related to this one. I did not get any solution.

```
2024-07-12 00:36:05 2024-07-12 03:36:05+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 9.0.0-1.el9 started.
2024-07-12 00:36:06 2024-07-12 03:36:06+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-07-12 00:36:06 2024-07-12 03:36:06+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 9.0.0-1.el9 started.
2024-07-12 00:36:07 '/var/lib/mysql/mysql.sock' -> '/var/run/mysqld/mysqld.sock'
2024-07-12 00:36:07 2024-07-12T03:36:07.191761Z 0 [System] [MY-015015] [Server] MySQL Server - start.
2024-07-12 00:36:07 2024-07-12T03:36:07.396324Z 0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2024-07-12 00:36:07 2024-07-12T03:36:07.399562Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 9.0.0) starting as process 1
2024-07-12 00:36:07 2024-07-12T03:36:07.401119Z 0 [Warning] [MY-013242] [Server] --character-set-server: 'utf8' is currently an alias for the character set UTF8MB3, but will be an alias for UTF8MB4 in a future release. Please consider using UTF8MB4 in order to be unambiguous.
2024-07-12 00:36:07 2024-07-12T03:36:07.403408Z 0 [Warning] [MY-010159] [Server] Setting lower_case_table_names=2 because file system for /var/lib/mysql/ is case insensitive
2024-07-12 00:36:07 2024-07-12T03:36:07.416298Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-07-12 00:36:08 2024-07-12T03:36:08.504374Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2024-07-12 00:36:08 mysqld: Table 'mysql.plugin' doesn't exist
2024-07-12 00:36:08 2024-07-12T03:36:08.650470Z 0 [ERROR] [MY-010735] [Server] Could not open the mysql.plugin table. Please perform the MySQL upgrade procedure.
2024-07-12 00:36:08 2024-07-12T03:36:08.650656Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 00:36:08 2024-07-12T03:36:08.650864Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 00:36:08 2024-07-12T03:36:08.651103Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 00:36:08 2024-07-12T03:36:08.651414Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 00:36:08 2024-07-12T03:36:08.651610Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 00:36:08 2024-07-12T03:36:08.651811Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 00:36:08 2024-07-12T03:36:08.652023Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 00:36:08 2024-07-12T03:36:08.728835Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-07-12 00:36:08 2024-07-12T03:36:08.806526Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-07-12 00:36:08 2024-07-12T03:36:08.827787Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2024-07-12 00:36:08 2024-07-12T03:36:08.828214Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2024-07-12 00:36:08 2024-07-12T03:36:08.843134Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
2024-07-12 00:36:08 2024-07-12T03:36:08.843454Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 00:36:08 2024-07-12T03:36:08.843746Z 0 [ERROR] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-001146 - Table 'mysql.component' doesn't exist
2024-07-12 00:36:08 2024-07-12T03:36:08.843874Z 0 [Warning] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-003543 - The mysql.component table is missing or has an incorrect definition.
2024-07-12 00:36:08 2024-07-12T03:36:08.844054Z 0 [ERROR] [MY-000067] [Server] unknown variable 'mysql_native_password=on else echo default-authentication-plugin=mysql_native_password'.
2024-07-12 00:36:08 2024-07-12T03:36:08.845648Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-07-12 00:36:10 2024-07-12T03:36:10.016387Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 9.0.0)  MySQL Community Server - GPL.
2024-07-12 00:36:10 2024-07-12T03:36:10.016420Z 0 [System] [MY-015016] [Server] MySQL Server - end.
```

+1

I met same issue

```
2024-07-12 17:31:20 2024-07-12 08:31:20+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 9.0.0-1.el9 started.
2024-07-12 17:31:20 2024-07-12 08:31:20+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-07-12 17:31:20 2024-07-12 08:31:20+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 9.0.0-1.el9 started.
2024-07-12 17:31:20 '/var/lib/mysql/mysql.sock' -> '/var/run/mysqld/mysqld.sock'
2024-07-12 17:31:20 2024-07-12T08:31:20.427629Z 0 [System] [MY-015015] [Server] MySQL Server - start.
2024-07-12 17:31:20 2024-07-12T08:31:20.571059Z 0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2024-07-12 17:31:20 2024-07-12T08:31:20.572394Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 9.0.0) starting as process 1
2024-07-12 17:31:20 2024-07-12T08:31:20.573325Z 0 [Warning] [MY-013242] [Server] --character-set-server: 'utf8' is currently an alias for the character set UTF8MB3, but will be an alias for UTF8MB4 in a future release. Please consider using UTF8MB4 in order to be unambiguous.
2024-07-12 17:31:20 2024-07-12T08:31:20.574105Z 0 [Warning] [MY-010159] [Server] Setting lower_case_table_names=2 because file system for /var/lib/mysql/ is case insensitive
2024-07-12 17:31:20 2024-07-12T08:31:20.577308Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-07-12 17:31:20 2024-07-12T08:31:20.674816Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2024-07-12 17:31:20 2024-07-12T08:31:20.800341Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2024-07-12 17:31:20 2024-07-12T08:31:20.800394Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2024-07-12 17:31:20 2024-07-12T08:31:20.803132Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
2024-07-12 17:31:20 2024-07-12T08:31:20.804778Z 0 [ERROR] [MY-000067] [Server] unknown variable 'mysql_native_password=on'.
2024-07-12 17:31:20 2024-07-12T08:31:20.805138Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-07-12 17:31:21 2024-07-12T08:31:21.866064Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 9.0.0)  MySQL Community Server - GPL.
2024-07-12 17:31:21 2024-07-12T08:31:21.866082Z 0 [System] [MY-015016] [Server] MySQL Server - end.
```

> > > I also had the same issue. Running MySql 8.0.32, Docker Desktop 4.32.0, on Mac OS.
> > > The if evaluation wasn't running correctly, making this change fixed it for me:
> > > ```
> > > # RUN if [ ${MYSQL_VERSION} > '8.4.0-0.000' ]; then \
> > > #   echo 'mysql_native_password=on' >> /etc/mysql/conf.d/my.cnf \
> > > # else \
> > > #   echo 'default-authentication-plugin=mysql_native_password' >> /etc/mysql/conf.d/my.cnf \
> > > # ;fi
> > > 
> > > RUN echo 'default-authentication-plugin=mysql_native_password' >> /etc/mysql/conf.d/my.cnf
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Obviously this isn't a good long term solution (if I ever wanted to upgrade past 8.4 I would manually have to change it again).
> > 
> > 
> > I am using `MYSQL_VERSION=latest` **(9.0)** and the patch notes for this version say:
> > `The mysql_native_password authentication plugin, deprecated in MySQL 8.0, has been removed, and the server now rejects mysql_native authentication requests from older client programs which do not have CLIENT_PLUGIN_AUTH capability. For backward compatibility, mysql_native_password remains available on the client; the client-side built-in authentication plugin has been converted into a dynamically loadable plugin.`
> > So trying to modify or do anything with those variables has not solved anything (at least for those versions), this is due to something more internal perhaps....
> 
> The if statement that's breaking is just setting those variables dependant on what MySql version you're running. So I removed the if, and just ran the line that sets the correct variable for MySql >8.4.
> 
> If you're running MySql >8.4 then of course you'll need to switch out that final line for the first part of the if statement..
> 
> I haven't tested it but this should work for MySql >8.4:
> 
> ```
> # RUN if [ ${MYSQL_VERSION} > '8.4.0-0.000' ]; then \
> #   echo 'mysql_native_password=on' >> /etc/mysql/conf.d/my.cnf \
> # else \
> #   echo 'default-authentication-plugin=mysql_native_password' >> /etc/mysql/conf.d/my.cnf \
> # ;fi
> 
> RUN echo 'mysql_native_password=on' >> /etc/mysql/conf.d/my.cnf
> ```

The problem remains the same in relation to these variables.....

```2024-07-12 10:08:44 2024-07-12 13:08:44+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 9.0.0-1.el9 started.
2024-07-12 10:08:44 2024-07-12 13:08:44+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-07-12 10:08:44 2024-07-12 13:08:44+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 9.0.0-1.el9 started.
2024-07-12 10:08:44 '/var/lib/mysql/mysql.sock' -> '/var/run/mysqld/mysqld.sock'
2024-07-12 10:08:45 2024-07-12T13:08:44.959015Z 0 [System] [MY-015015] [Server] MySQL Server - start.
2024-07-12 10:08:45 2024-07-12T13:08:45.188214Z 0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2024-07-12 10:08:45 2024-07-12T13:08:45.193717Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 9.0.0) starting as process 1
2024-07-12 10:08:45 2024-07-12T13:08:45.195385Z 0 [Warning] [MY-013242] [Server] --character-set-server: 'utf8' is currently an alias for the character set UTF8MB3, but will be an alias for UTF8MB4 in a future release. Please consider using UTF8MB4 in order to be unambiguous.
2024-07-12 10:08:45 2024-07-12T13:08:45.198766Z 0 [Warning] [MY-010159] [Server] Setting lower_case_table_names=2 because file system for /var/lib/mysql/ is case insensitive
2024-07-12 10:08:45 2024-07-12T13:08:45.213339Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-07-12 10:08:46 2024-07-12T13:08:46.633803Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2024-07-12 10:08:46 mysqld: Table 'mysql.plugin' doesn't exist
2024-07-12 10:08:46 2024-07-12T13:08:46.867878Z 0 [ERROR] [MY-010735] [Server] Could not open the mysql.plugin table. Please perform the MySQL upgrade procedure.
2024-07-12 10:08:46 2024-07-12T13:08:46.868190Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 10:08:46 2024-07-12T13:08:46.868410Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 10:08:46 2024-07-12T13:08:46.868602Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 10:08:46 2024-07-12T13:08:46.868824Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 10:08:46 2024-07-12T13:08:46.893717Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 10:08:46 2024-07-12T13:08:46.894090Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 10:08:46 2024-07-12T13:08:46.894297Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 10:08:46 2024-07-12T13:08:46.997819Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-07-12 10:08:47 2024-07-12T13:08:47.101999Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-07-12 10:08:47 2024-07-12T13:08:47.122953Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2024-07-12 10:08:47 2024-07-12T13:08:47.123456Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2024-07-12 10:08:47 2024-07-12T13:08:47.137279Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
2024-07-12 10:08:47 2024-07-12T13:08:47.137691Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-07-12 10:08:47 2024-07-12T13:08:47.137945Z 0 [ERROR] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-001146 - Table 'mysql.component' doesn't exist
2024-07-12 10:08:47 2024-07-12T13:08:47.137989Z 0 [Warning] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-003543 - The mysql.component table is missing or has an incorrect definition.
2024-07-12 10:08:47 2024-07-12T13:08:47.138078Z 0 [ERROR] [MY-000067] [Server] unknown variable 'mysql_native_password=on'.
2024-07-12 10:08:47 2024-07-12T13:08:47.138858Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-07-12 10:08:48 2024-07-12T13:08:48.269973Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 9.0.0)  MySQL Community Server - GPL.
2024-07-12 10:08:48 2024-07-12T13:08:48.270005Z 0 [System] [MY-015016] [Server] MySQL Server - end.```

Tried all of the solutions in here , I still get the same issue , mysql just exit everytime I start it 

`2024-07-19 14:21:13 2024-07-19 06:21:13+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.38-1.el9 started.
2024-07-19 14:21:15 2024-07-19 06:21:15+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-07-19 14:21:15 2024-07-19 06:21:15+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.38-1.el9 started.
2024-07-19 14:21:15 '/var/lib/mysql/mysql.sock' -> '/var/run/mysqld/mysqld.sock'
2024-07-19 14:21:15 2024-07-19T06:21:15.768563Z 0 [Warning] [MY-011068] [Server] The syntax '--skip-host-cache' is deprecated and will be removed in a future release. Please use SET GLOBAL host_cache_size=0 instead.
2024-07-19 14:21:15 2024-07-19T06:21:15.768636Z 0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2024-07-19 14:21:15 2024-07-19T06:21:15.777061Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.38) starting as process 1
2024-07-19 14:21:15 2024-07-19T06:21:15.778731Z 0 [Warning] [MY-013242] [Server] --character-set-server: 'utf8' is currently an alias for the character set UTF8MB3, but will be an alias for UTF8MB4 in a future release. Please consider using UTF8MB4 in order to be unambiguous.
2024-07-19 14:21:15 2024-07-19T06:21:15.785889Z 0 [Warning] [MY-010159] [Server] Setting lower_case_table_names=2 because file system for /var/lib/mysql/ is case insensitive
2024-07-19 14:21:15 2024-07-19T06:21:15.816982Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-07-19 14:21:18 2024-07-19T06:21:18.473797Z 1 [ERROR] [MY-014061] [InnoDB] Invalid MySQL server downgrade: Cannot downgrade from 80100 to 80038. Downgrade is only permitted between patch releases.
2024-07-19 14:21:18 mysqld: Can't open file: 'mysql.ibd' (errno: 0 - )
2024-07-19 14:21:18 2024-07-19T06:21:18.564439Z 1 [ERROR] [MY-010334] [Server] Failed to initialize DD Storage Engine
2024-07-19 14:21:18 2024-07-19T06:21:18.564825Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
2024-07-19 14:21:18 2024-07-19T06:21:18.564907Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-07-19 14:21:18 2024-07-19T06:21:18.566651Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.38)  MySQL Community Server - GPL.`

--- a/.env.example
+++ b/.env.example
@@ -407,7 +407,8 @@ APACHE_FOR_MAC_M1=false
 
 ### MYSQL #################################################
 
-MYSQL_VERSION=latest
+# 5.7, 8.0, 8.4, 9.0
+MYSQL_VERSION=8.4
 MYSQL_DATABASE=default
 MYSQL_USER=default
 MYSQL_PASSWORD=secret

--- a/mysql/Dockerfile
+++ b/mysql/Dockerfile
@@ -1,8 +1,6 @@
 ARG MYSQL_VERSION
 FROM mysql:${MYSQL_VERSION}
 
-LABEL maintainer="Mahmoud Zalt <mahmoud@zalt.me>"
-
 #####################################
 # Set Timezone
 #####################################
@@ -15,10 +13,6 @@ COPY my.cnf /etc/mysql/conf.d/my.cnf
 
 RUN chmod 0444 /etc/mysql/conf.d/my.cnf
 
-RUN if [ ${MYSQL_VERSION} > '8.4.0-0.000' ]; then \
-  echo 'mysql_native_password=on' >> /etc/mysql/conf.d/my.cnf \
-;else \
-  echo 'default-authentication-plugin=mysql_native_password' >> /etc/mysql/conf.d/my.cnf \
-;fi
-
-EXPOSE 3306
+RUN if [ ${MYSQL_MAJOR} = '8.0' ]; then \
+    echo 'default-authentication-plugin=mysql_native_password' >> /etc/mysql/conf.d/my.cnf; \
+  fi

